libvcx/src/api_lib/api_c/out_of_band.rs

```diff
@@ -0,0 +1,450 @@
+use std::ptr;
+use libc::c_char;
+
+use aries_vcx::indy_sys::CommandHandle;
+
+use crate::api_lib::api_handle::out_of_band;
+use crate::api_lib::utils::cstring::CStringUtils;
+use crate::api_lib::utils::runtime::execute;
+use crate::error::prelude::*;
+use aries_vcx::utils::error;
+
+#[no_mangle]
+pub extern fn vcx_out_of_band_create(command_handle: CommandHandle,
+                                     config: *const c_char,
+                                     cb: Option<extern fn(xcommand_handle: CommandHandle, err: u32, handle: u32)>) -> u32 {
+    info!("vcx_out_of_band_create >>>");
+
+    check_useful_c_str!(config, VcxErrorKind::InvalidOption);
+    check_useful_c_callback!(cb, VcxErrorKind::InvalidOption);
+
+    trace!("vcx_out_of_band_create(command_handle: {}, config: {})", command_handle, config);
+
+    execute(move || {
+        match out_of_band::create_out_of_band(&config) {
+            Ok(handle) => {
+                trace!("vcx_out_of_band_create_cb(command_handle: {}, rc: {}, handle: {})",
+                       command_handle, error::SUCCESS.message, handle);
+                cb(command_handle, error::SUCCESS.code_num, handle);
+            }
+            Err(x) => {
+                warn!("vcx_out_of_band_create_cb(command_handle: {}, rc: {}, handle: {})",
+                      command_handle, x, 0);
+                cb(command_handle, x.into(), 0);
+            }
+        }
+        Ok(())
+    });
+
+    error::SUCCESS.code_num
+}
+
+#[no_mangle]
+pub extern fn vcx_out_of_band_create_from_message(command_handle: CommandHandle,
+                                                  message: *const c_char,
+                                                  cb: Option<extern fn(xcommand_handle: CommandHandle, err: u32, handle: u32)>) -> u32 {
+    info!("vcx_out_of_band_create_from_message >>>");
+
+    check_useful_c_str!(message, VcxErrorKind::InvalidOption);
+    check_useful_c_callback!(cb, VcxErrorKind::InvalidOption);
+
+    trace!("vcx_out_of_band_create_from_message(command_handle: {}, message: {})", command_handle, message);
+
+    execute(move || {
+        match out_of_band::create_out_of_band_msg_from_msg(&message) {
+            Ok(handle) => {
+                trace!("vcx_out_of_band_create_from_message_cb(command_handle: {}, rc: {}, handle: {})",
+                       command_handle, error::SUCCESS.message, handle);
+                cb(command_handle, error::SUCCESS.code_num, handle);
+            }
+            Err(x) => {
+                warn!("vcx_out_of_band_create_from_message_cb(command_handle: {}, rc: {}, handle: {}):",
+                      command_handle, x, 0);
+                cb(command_handle, x.into(), 0);
+            }
+        }
+        Ok(())
+    });
+
+    error::SUCCESS.code_num
+}
+
+#[no_mangle]
+pub extern fn vcx_out_of_band_append_message(command_handle: CommandHandle,
```

Comment:
same as above, also applies for next couple methods, I'd distinguish them with `_sender_` and `_receiver_` unless shared for both